WordPress/Classes/ViewRelated/Comments/CommentDetailViewController.swift

```diff
@@ -884,4 +892,63 @@ private extension CommentDetailViewController {
         // TODO: create reply
     }
 
+    func configureSuggestionsView() {
+        guard shouldShowSuggestions,
+              let siteID = siteID,
+        let replyTextView = replyTextView else {
+            return
+        }
+
+        let suggestionsView = SuggestionsTableView(siteID: siteID, suggestionType: .mention, delegate: self)
+        suggestionsView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(suggestionsView)
+
+        NSLayoutConstraint.activate([
+            suggestionsView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            suggestionsView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            suggestionsView.topAnchor.constraint(equalTo: view.topAnchor),
+            suggestionsView.bottomAnchor.constraint(equalTo: replyTextView.topAnchor)
+        ])
+
+        suggestionsTableView = suggestionsView
+    }
+
+    var shouldShowSuggestions: Bool {
+        guard let siteID = siteID,
+              let blog = Blog.lookup(withID: siteID, in: ContextManager.shared.mainContext) else {
+                  return false
+              }
+
+        return SuggestionService.shared.shouldShowSuggestions(for: blog)
+    }
+
+}
+
+// MARK: - ReplyTextViewDelegate
+
+extension CommentDetailViewController: ReplyTextViewDelegate {
+
+    func textView(_ textView: UITextView, didTypeWord word: String) {
+        suggestionsTableView?.showSuggestions(forWord: word)
+    }
+
+    func replyTextView(_ replyTextView: ReplyTextView, willEnterFullScreen controller: FullScreenCommentReplyViewController) {
+        suggestionsTableView?.hideSuggestions()
+
+        if let siteID = siteID {
+            controller.enableSuggestions(with: siteID)
+        }
+    }
+
+}
+
+// MARK: - SuggestionsTableViewDelegate
+
+extension CommentDetailViewController: SuggestionsTableViewDelegate {
+
+    func suggestionsTableView(_ suggestionsTableView: SuggestionsTableView, didSelectSuggestion suggestion: String?, forSearchText text: String) {
+        replyTextView?.replaceTextAtCaret(text as NSString?, withText: suggestion)
+        suggestionsTableView.showSuggestions(forWord: String())
```

Comment:
What do you think about calling `hideSuggestions()` here instead? At first, I assumed that calling `showSuggestions` with an empty string means showing a list of all mentionable users, but turns out it's just to dismiss the suggestion view. But then I checked that `hideSuggestions()` is actually doing the same:
```Objective-C
- (void)hideSuggestions
{
    [self showSuggestionsForWord:@""];
}
```

So maybe we could use `hideSuggestions()` instead for clarity? 😄

Reply:
Ha! That's what I get for copy + paste + assuming it was right to start with. Can do! Thanks.